.github/workflows/docker-ci-tools.yml

```diff
@@ -0,0 +1,76 @@
+name: docker-ci-tools
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - 'tools/**'
+
+env:
+  IMAGE_NAME: grafana/tempo-ci-tools
+
+# Needed to login to DockerHub
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+
+  get-image-tag:
+    runs-on: ubuntu-24.04
+    outputs:
+      tag: ${{ steps.get-tag.outputs.tag }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - id: get-tag
+        run: |
+          echo "tag=$(./tools/image-tag)" >> "$GITHUB_OUTPUT"
+
+  docker-ci-tools:
+    needs: get-image-tag
+    strategy:
+      matrix:
+        runner_arch: [ { runner: ubuntu-24.04, arch: amd64 }, { runner: github-hosted-ubuntu-arm64, arch: arm64 } ]
+    runs-on: ${{ matrix.runner_arch.runner }}
+    env:
+      TAG: ${{ needs.get-image-tag.outputs.tag }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Login to DockerHub
+        uses: grafana/shared-workflows/actions/dockerhub-login@dockerhub-login-v1.0.0
+
+      - name: docker-build-and-push
+        run: |
+          TAG_ARCH="$TAG-${{ matrix.runner_arch.arch }}"
+          docker build -f tools/Dockerfile -t $IMAGE_NAME:$TAG_ARCH .
+          docker push $IMAGE_NAME:$TAG_ARCH
+
+  manifest:
```

Comment:
I think this is not needed since we are not pushing a multi-architecture image yet

Reply:
And it can be achieved as well with  docker `build-push-action `

Reply:
In Drone we do something very similar to what I'm doing here, that is, build each platform natively and then create a multi-platform manifest -- only that in Drone we use some opaque plugins.

To clarify, this workflow is a combination of these Drone pipelines:
- https://github.com/grafana/tempo/blob/main/.drone/drone.yml#L217
- https://github.com/grafana/tempo/blob/main/.drone/drone.yml#L250
- https://github.com/grafana/tempo/blob/main/.drone/drone.yml#L285